targets/TARGET_STM/can_api.c

```diff
@@ -235,8 +275,34 @@ int can_frequency(can_t *obj, int f)
         error("HAL_FDCAN_Stop error\n");
     }
 
-    /* See can_init_freq function for calculation details */
-    int ntq = 10000000 / f;
+
+    /* See can_init_freq function for calculation details
+     *
+     * !Attention Not all bitrates can be covered with all fdcan-core-clk values. When a clk
+     * does not work for the desired bitrate, change system_clock settings for FDCAN_CLK
+     * (default FDCAN_CLK is PLLQ)
+     */
+#ifdef TARGET_STM32G4
+    int ntq = HAL_RCCEx_GetPeriphCLKFreq(RCC_PERIPHCLK_FDCAN) / f;
+#else
+    // STM32H7 doesn't support yet HAL_RCCEx_GetPeriphCLKFreq for FDCAN
+    // Internal ST ticket 92465
+    int ntq = 10000000 / hz;
```

Comment:
Not hz in this function but f...

Reply:
Sorry for that slip-up, I have no H7 to test, but should have noticed it anyway. Thanks for testing!

Reply:
Changed in 73493b9